tools/rdm/TestMixins.py

```diff
@@ -853,28 +909,46 @@ def Test(self):
     self.SendSet(ROOT_DEVICE, self.pid, [settings_supported + 1])
 
 class GetSettingDescriptionsMixin(object):
-  """Perform a GET for each setting in the range 0 .. NumberOfSettings().
+  """Perform a GET for each setting in a range.
+
+    The range can either be an array of settings, which don't need to be 
+    sequential, or a count, in which case it will check
+    FIRST_INDEX_OFFSET to FIRST_INDEX_OFFSET + count.
 
     Subclasses must define EXPECTED_FIELD, which is the field to validate the
     index against and DESCRIPTION_FIELD, which is the field to check for
     unprintable characters.
+
+    If ALLOWED_NACK is defined, this adds a custom NackGetResult to the list of
+    allowed results.
   """
   CATEGORY = TestCategory.DIMMER_SETTINGS
+  ALLOWED_NACK = None
+  FIRST_INDEX_OFFSET = 1
 
-  def NumberOfSettings(self):
+  def ListOfSettings(self):
     # By default we use the first property from REQUIRES
-    return self.Property(self.REQUIRES[0])
+    if isinstance(self.Property(self.REQUIRES[0]), list):
```

Comment:
The implicit behavior here is confusing to begin with and this makes it harder.

How about we say that each test implements the ListOfSettings() method. Then many of them can be 

def ListOfSettings(self):
  return self.Property(self.REQUIRES[0])

and some can be:

def ListOfSettings(self):
  max_count = self.Property(self.REQUIRES[0])
  if max_count is None:
    return []
  return xrange(0, max_count)

I think it's better to be explicit about these, even if it is slightly more code.